lib/postgrex_wal/pg_source.ex

```diff
@@ -194,4 +196,41 @@ defmodule PostgrexWal.PgSource do
 
   @epoch DateTime.to_unix(~U[2000-01-01 00:00:00Z], :microsecond)
   defp current_time, do: System.os_time(:microsecond) - @epoch
+
+  # ?A => StreamAbort,
+  # ?B => Begin,
+  # ?C => Commit,
+  # ?D => Delete,
+  # ?E => StreamStop,
+  # ?I => Insert,
+  # ?M => Message,
+  # ?O => Origin,
+  # ?R => Relation,
+  # ?S => StreamStart,
+  # ?T => Truncate,
+  # ?U => Update,
+  # ?Y => Type,
+  # ?c => StreamCommit
+  defp in_stream_preprocess(<<key::8, _rest::binary>> = payload, state) do
+    in_stream? =
+      case key do
+        ?S ->
+          state.in_stream && Logger.error("stream flag consecutively true")
+          true
+
+        ?E ->
+          state.in_stream || Logger.error("stream flag consecutively false")
+          false
+
+        _ ->
+          state.in_stream
+      end
+
+    payload =
+      if in_stream? and key in [?D, ?I, ?M, ?R, ?T, ?U, ?Y],
+        do: {:in_stream, payload},
+        else: payload
+
+    {payload, %{state | in_stream: in_stream?}}
```

Comment:
这个看你想不想改， `:in_stream?` 作为 key 是合法的
```suggestion
    {payload, %{state | in_stream?: in_stream?}}
```

Reply:
原本是这样写的，后面纠结了 一下，去掉了`?`， 之所以纠结这个`？后缀`命名，主要是担心：
1. `{payload, %{state | in_stream?: in_stream?}}` 这样的`key`不美观，在其它`repository`代码中少见。
2. 怕与变量`in_stream?` 混淆。

ok，已经更改。
Done